Sat Mar 18 17:30:42 CET 2006  VMiklos <vmiklos@frugalware.org>
  * pmdb_t: allow repo names to be longer than 128 chars
diff -rN -u old-pacman/lib/libalpm/db.c new-pacman/lib/libalpm/db.c
--- old-pacman/lib/libalpm/db.c	2006-03-18 18:02:22.000000000 +0100
+++ new-pacman/lib/libalpm/db.c	2006-03-18 18:02:22.000000000 +0100
@@ -57,7 +57,7 @@
 	}
 	sprintf(db->path, "%s%s/%s", root, dbpath, treename);
 
-	STRNCPY(db->treename, treename, DB_TREENAME_LEN);
+	STRNCPY(db->treename, treename, PATH_MAX);
 
 	db->pkgcache = NULL;
 	db->grpcache = NULL;
diff -rN -u old-pacman/lib/libalpm/db.h new-pacman/lib/libalpm/db.h
--- old-pacman/lib/libalpm/db.h	2006-03-18 18:02:22.000000000 +0100
+++ new-pacman/lib/libalpm/db.h	2006-03-18 18:02:22.000000000 +0100
@@ -21,6 +21,7 @@
 #ifndef _ALPM_DB_H
 #define _ALPM_DB_H
 
+#include <limits.h>
 #include "package.h"
 
 /* Database entries */
@@ -31,14 +32,12 @@
 #define INFRQ_SCRIPLET 0x08
 #define INFRQ_ALL      0xFF
 
-#define DB_TREENAME_LEN 128
-
 #define DB_O_CREATE 0x01
 
 /* Database */
 typedef struct __pmdb_t {
 	char *path;
-	char treename[DB_TREENAME_LEN];
+	char treename[PATH_MAX];
 	void *handle;
 	PMList *pkgcache;
 	PMList *grpcache;

